File: files/js/src/main/scala/com/swoval/files/Filter.scala
package com.swoval.files

/**
 * Generic Filter functional interface. Primarily used by [[QuickList]].
 * @tparam T The type of object to filter
 */
trait Filter[T] {

  /**
   * Accept only some instances of {@code T}.
   * @param t The instance to filter
   * @return true if the instance is accepted
   */
  def accept(t: T): Boolean

}

File: files/jvm/src/main/java/com/swoval/functional/Filter.java
package com.swoval.functional;

/**
 * Generic Filter functional interface.
 *
 * @param <T> The type of object to filter
 */
public interface Filter<T> {

  /**
   * Accept only some instances of {@code T}.
   *
   * @param t The instance to filter
   * @return true if the instance is accepted
   */
  boolean accept(final T t);
}
